extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/model/TradfriVersion.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.tradfri.internal.model;
+
+import org.eclipse.jdt.annotation.NonNull;
+
+/**
+ * The {@link TradfriVersion} class is a default implementation of comparing TRÅDFRI versions.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+public class TradfriVersion implements Comparable<TradfriVersion> {
+    private static final String VERSION_PATTERN = "[0-9]+(\\.[0-9]+)*";
+    private static final String VERSION_DELIMITER = "\\.";
+    private final String version;
+
+    /**
+     * Create a new instance.
+     *
+     * @param version the version string
+     */
+    public TradfriVersion(@NonNull String version) {
+        if (version == null)
+            throw new IllegalArgumentException("TradfriVersion cannot be created as version is null.");
+        if (!version.matches(VERSION_PATTERN))
```

Comment:
As you already dereference the version object I don't think you need to check above for null and throw a IAE. A NPE is also okay if you annotate that a non-null reference should be given.
Both are runtime exceptions, any reason to add a check and throw an IAE instead of the NPE here?